frontend/src/console/components/Hero/Hero.tsx

```diff
@@ -47,39 +46,59 @@ export class HeroComponent extends React.PureComponent<Props, State> {
   constructor(props: Props) {
     super(props);
 
-    this.state = {
-      title: '',
-      subtitle: ''
-    };
+    this.state = this.getHeroContents(props);
   }
 
-  public componentDidMount() {
-    this.checkAndUpdateState();
-  }
+  public componentDidUpdate(prevProps: Props) {
+    const oldState = this.getHeroContents(prevProps);
+    const newState = this.getHeroContents(this.props);
 
-  public componentDidUpdate() {
-    this.checkAndUpdateState();
+    // Only update internal state if something changed after props updated.
```

Comment:
There is an broken edge case here - if you set title/subtitle to `reset_default` twice, it will stay like this.

Reply:
I'll revert the `reset_default` thing back to how it worked before, and check if there's any other edge case.

Reply:
@giovannicimolin, I already tested this and there is a similar one - when you remove everything from this field twice.

Reply:
Reverted back to @lgp171188 initial implementation. 
I've spend quite some time trying to fix this state management issue already.

Reply:
I feel that your approach was actually more user-friendly :P
Though it's not in the scope of this ticket.